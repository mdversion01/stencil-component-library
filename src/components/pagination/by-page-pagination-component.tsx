// src/components/pagination/by-page-pagination-component.tsx
 import { Component, Prop, Event, EventEmitter, h, Element, Fragment } from '@stencil/core';

@Component({
  tag: 'by-page-pagination-component',
  styleUrls: ['./pagination-styles.scss', '../input-field/input-field-styles.scss', '../plumage-input-field/plumage-input-field-styles.scss', '../form-styles.scss'],
  shadow: false,
})
export class ByPagePagination {
  @Element() el!: HTMLElement;

  @Prop() currentPage: number = 1;
  @Prop() totalPages: number = 1;
  @Prop() goToButtons: string = '';
  @Prop() size: '' | 'sm' | 'lg' = '';
  @Prop() paginationLayout: '' | 'center' | 'end' = '';
  @Prop() plumage: boolean = false;
  @Prop({ attribute: 'control-id' }) controlId?: string;

  @Event({ eventName: 'change-page' }) changePage!: EventEmitter<{ page: number }>;

  private isInputFocused = false;

  connectedCallback() {
    document.addEventListener('click', this.onDocumentClick);
  }
  disconnectedCallback() {
    document.removeEventListener('click', this.onDocumentClick);
  }

  private onDocumentClick = () => {
    if (!this.isInputFocused) {
      const bFocusDiv = this.el?.querySelector('.b-focus') as HTMLElement | null;
      if (bFocusDiv) {
        bFocusDiv.style.width = '0';
        bFocusDiv.style.left = '50%';
      }
    }
  };

  private firstPage = () => this.changePage.emit({ page: 1 });
  private prevPage = () => {
    if (this.currentPage > 1) this.changePage.emit({ page: this.currentPage - 1 });
  };
  private nextPage = () => {
    if (this.currentPage < this.totalPages) this.changePage.emit({ page: this.currentPage + 1 });
  };
  private lastPage = () => this.changePage.emit({ page: this.totalPages });

  private onInputChange = (e: Event) => {
    const target = e.target as HTMLInputElement;
    const newPage = parseInt(target?.value || '', 10);
    if (!Number.isNaN(newPage) && newPage >= 1 && newPage <= this.totalPages) this.changePage.emit({ page: newPage });
  };

  private onFocus = () => {
    this.isInputFocused = true;
    const bFocusDiv = this.el?.querySelector('.b-focus') as HTMLElement | null;
    if (bFocusDiv) {
      bFocusDiv.style.width = '100%';
      bFocusDiv.style.left = '0';
    }
  };

  private onBlur = () => {
    this.isInputFocused = false;
    setTimeout(() => {
      if (!this.isInputFocused) {
        const bFocusDiv = this.el?.querySelector('.b-focus') as HTMLElement | null;
        if (bFocusDiv) {
          bFocusDiv.style.width = '0';
          bFocusDiv.style.left = '50%';
        }
      }
    }, 100);
  };

  render() {
    const sizeCls = this.size === 'sm' ? ' pagination-sm' : this.size === 'lg' ? ' pagination-lg' : '';
    const layoutCls = this.paginationLayout === 'center' ? ' justify-content-center' : this.paginationLayout === 'end' ? ' justify-content-end' : '';
    const plumageCls = this.plumage ? ' plumage' : '';
    const basicInputSize = this.size === 'sm' ? ' basic-input-sm' : this.size === 'lg' ? ' basic-input-lg' : '';
    const plInputSize = this.size === 'sm' ? ' pl-input-sm' : this.size === 'lg' ? ' pl-input-lg' : '';
    const controls = this.controlId ?? this.el.id;

    return (
      <ul role="menubar" aria-disabled="false" aria-label="Pagination" class={`pagination by-page b-pagination${sizeCls}${layoutCls}${plumageCls}`}>
        <li role="presentation" aria-hidden={this.currentPage === 1} class={`page-item${this.currentPage === 1 ? ' disabled' : ''}`}>
          <button
            role="menuitem"
            type="button"
            tabIndex={this.currentPage === 1 ? -1 : 0}
            aria-label="Go to first page"
            aria-controls={controls}
            class="page-link"
            onClick={this.firstPage}
            disabled={this.currentPage === 1}
          >
            {this.goToButtons === 'text' ? 'First' : '«'}
          </button>
        </li>

        <li role="presentation" aria-hidden={this.currentPage === 1} class={`page-item${this.currentPage === 1 ? ' disabled' : ''}`}>
          <button
            role="menuitem"
            type="button"
            tabIndex={this.currentPage === 1 ? -1 : 0}
            aria-label="Go to previous page"
            aria-controls={controls}
            class="page-link"
            onClick={this.prevPage}
            disabled={this.currentPage === 1}
          >
            {this.goToButtons === 'text' ? 'Prev' : '‹'}
          </button>
        </li>

        <li role="presentation" aria-hidden="true" class="page-item">
          <div class="pages">
            Page{' '}
            {this.plumage ? (
              <div class="pl-input-container page-input-wrapper" role="presentation" aria-labelledby="pageNumberField">
                <label class="sr-only" htmlFor="pageNumberField">
                  Page number
                </label>
                <input
                  type="text"
                  class={`pl-form-control page-input${plInputSize}`}
                  aria-label="Page number"
                  inputMode="numeric"
                  pattern="[0-9]*"
                  id="paginationInput"
                  aria-labelledby="paginationInputLabel paginationDescription"
                  value={String(this.currentPage)}
                  onChange={this.onInputChange}
                  onFocus={this.onFocus}
                  onBlur={this.onBlur}
                />
                <div class="b-underline" role="presentation">
                  <div class="b-focus" role="presentation" aria-hidden="true" />
                </div>
              </div>
             ) : (
              <Fragment>
                <label class="sr-only" htmlFor="pageNumberField">
                  Page number
                </label>
                <input
                  type="text"
                  inputMode="numeric"
                  pattern="[0-9]*"
                  id="paginationInput"
                  aria-label="Page number"
                  aria-labelledby="paginationInputLabel paginationDescription"
                  class={`form-control page-input${basicInputSize}`}
                  value={String(this.currentPage)}
                  onChange={this.onInputChange}
                />
              </Fragment>
            )}{' '}
            of {this.totalPages}
          </div>
        </li>

        <li role="presentation" aria-hidden={this.currentPage === this.totalPages} class={`page-item${this.currentPage === this.totalPages ? ' disabled' : ''}`}>
          <button
            role="menuitem"
            type="button"
            tabIndex={this.currentPage === this.totalPages ? -1 : 0}
            aria-label="Go to next page"
            aria-controls={controls}
            class="page-link"
            onClick={this.nextPage}
            disabled={this.currentPage === this.totalPages}
          >
            {this.goToButtons === 'text' ? 'Next' : '›'}
          </button>
        </li>

        <li role="presentation" aria-hidden={this.currentPage === this.totalPages} class={`page-item${this.currentPage === this.totalPages ? ' disabled' : ''}`}>
          <button
            role="menuitem"
            type="button"
            tabIndex={this.currentPage === this.totalPages ? -1 : 0}
            aria-label="Go to last page"
            aria-controls={controls}
            class="page-link"
            onClick={this.lastPage}
            disabled={this.currentPage === this.totalPages}
          >
            {this.goToButtons === 'text' ? 'Last' : '»'}
          </button>
        </li>
      </ul>
    );
  }
}
