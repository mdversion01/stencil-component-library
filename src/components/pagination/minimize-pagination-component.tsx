// src/components/pagination/minimize-pagination-component.tsx
import { Component, Prop, Event, EventEmitter, h, Element } from '@stencil/core';

@Component({
  tag: 'minimize-pagination-component',
  styleUrls: ['./pagination-styles.scss'],
  shadow: false,
})
export class MinimizePagination {
  @Element() el!: HTMLElement;

  @Prop() currentPage: number = 1;
  @Prop() totalPages: number = 1;
  @Prop() goToButtons: string = '';
  @Prop() size: '' | 'sm' | 'lg' = '';
  @Prop() paginationLayout: '' | 'center' | 'end' = '';
  @Prop() plumage: boolean = false;
  // DO NOT use `id` here; use a different prop/attribute
  @Prop({ attribute: 'control-id' }) controlId?: string;

  @Event({ eventName: 'change-page' }) changePage!: EventEmitter<{ page: number }>;

  private firstPage = () => this.changePage.emit({ page: 1 });
  private prevPage = () => { if (this.currentPage > 1) this.changePage.emit({ page: this.currentPage - 1 }); };
  private nextPage = () => { if (this.currentPage < this.totalPages) this.changePage.emit({ page: this.currentPage + 1 }); };
  private lastPage = () => this.changePage.emit({ page: this.totalPages });

  render() {
    const sizeCls = this.size === 'sm' ? ' pagination-sm' : this.size === 'lg' ? ' pagination-lg' : '';
    const layoutCls = this.paginationLayout === 'center' ? ' justify-content-center'
                     : this.paginationLayout === 'end' ? ' justify-content-end' : '';
    const plumageCls = this.plumage ? ' plumage' : '';
    const controls = this.controlId ?? this.el.id; // fallback to host id

    return (
      <ul role="menubar" aria-disabled="false" aria-label="Pagination"
          class={`pagination b-pagination${sizeCls}${layoutCls}${plumageCls}`}>
        <li role="presentation" aria-hidden={this.currentPage === 1 as any}
            class={`page-item${this.currentPage === 1 ? ' disabled' : ''}`}>
          <button role="menuitem" type="button" tabIndex={0}
                  aria-label="Go to first page" aria-controls={controls}
                  class="page-link" onClick={this.firstPage}
                  disabled={this.currentPage === 1}>
            {this.goToButtons === 'text' ? 'First' : '«'}
          </button>
        </li>

        <li role="presentation" aria-hidden={this.currentPage === 1 as any}
            class={`page-item${this.currentPage === 1 ? ' disabled' : ''}`}>
          <button role="menuitem" type="button" tabIndex={0}
                  aria-label="Go to previous page" aria-controls={controls}
                  class="page-link" onClick={this.prevPage}
                  disabled={this.currentPage === 1}>
            {this.goToButtons === 'text' ? 'Prev' : '‹'}
          </button>
        </li>

        <li role="presentation" aria-hidden={this.currentPage === this.totalPages as any}
            class={`page-item${this.currentPage === this.totalPages ? ' disabled' : ''}`}>
          <button role="menuitem" type="button" tabIndex={0}
                  aria-label="Go to next page" aria-controls={controls}
                  class="page-link" onClick={this.nextPage}
                  disabled={this.currentPage === this.totalPages}>
            {this.goToButtons === 'text' ? 'Next' : '›'}
          </button>
        </li>

        <li role="presentation" aria-hidden={this.currentPage === this.totalPages as any}
            class={`page-item${this.currentPage === this.totalPages ? ' disabled' : ''}`}>
          <button role="menuitem" type="button" tabIndex={0}
                  aria-label="Go to last page" aria-controls={controls}
                  class="page-link" onClick={this.lastPage}
                  disabled={this.currentPage === this.totalPages}>
            {this.goToButtons === 'text' ? 'Last' : '»'}
          </button>
        </li>
      </ul>
    );
  }
}
