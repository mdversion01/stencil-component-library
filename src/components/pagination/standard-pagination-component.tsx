// src/components/pagination/standard-pagination-component.tsx
import { Component, Prop, Event, EventEmitter, h, Fragment } from '@stencil/core';

@Component({
  tag: 'standard-pagination-component',
  styleUrls: ['./pagination-styles.scss'],
  shadow: false,
})
export class StandardPagination {
  @Prop() currentPage: number = 1;
  @Prop() totalPages: number = 1;
  @Prop() hideGotoEndButtons: boolean = false;
  @Prop() goToButtons: string = '';
  @Prop() size: '' | 'sm' | 'lg' = '';
  @Prop() paginationLayout: '' | 'center' | 'end' | 'fill' | 'fill-left' | 'fill-right' = '';
  @Prop() hideEllipsis: boolean = false;
  @Prop() limit: number = 3;
  @Prop() paginationVariantColor: string = '';
  @Prop() plumage: boolean = false;

  @Event({ eventName: 'change-page' }) changePage!: EventEmitter<{ page: number }>;

  private emitChange(page: number) {
    this.changePage.emit({ page });
  }

  private nextPage = () => {
    if (this.currentPage < this.totalPages) this.emitChange(this.currentPage + 1);
  };
  private prevPage = () => {
    if (this.currentPage > 1) this.emitChange(this.currentPage - 1);
  };
  private firstPage = () => this.emitChange(1);
  private lastPage = () => this.emitChange(this.totalPages);

  private renderEllipsis() {
    if (this.hideEllipsis) return null;
    const liFlex =
      this.paginationLayout === 'fill' ||
      this.paginationLayout === 'fill-left' ||
      this.paginationLayout === 'fill-right'
        ? ' flex-fill d-flex'
        : '';
    const aFlex =
      this.paginationLayout === 'fill' ||
      this.paginationLayout === 'fill-left' ||
      this.paginationLayout === 'fill-right'
        ? ' flex-fill d-flex'
        : '';
    return (
      <li role="separator" class={`page-item disabled bv-d-xs-down-none${liFlex}`}>
        <span class={`page-link${aFlex}`}>...</span>
      </li>
    );
  }

  private renderPageButton(page: number) {
    const liFlex =
      this.paginationLayout === 'fill' ||
      this.paginationLayout === 'fill-left' ||
      this.paginationLayout === 'fill-right'
        ? ' flex-fill d-flex'
        : '';
    const aFlex =
      this.paginationLayout === 'fill' ||
      this.paginationLayout === 'fill-left' ||
      this.paginationLayout === 'fill-right'
        ? ' flex-grow-1'
        : '';
    const active = this.currentPage === page;
    return (
      <li
        role="presentation"
        class={`page-item${active ? ' active' : ''}${liFlex}`}
      >
        <button
          role="menuitemradio"
          type="button"
          aria-controls="pagination"
          aria-label={`Go to page ${page}`}
          aria-checked={active ? ('true' as any) : ('false' as any)}
          tabIndex={active ? -1 : 0}
          class={`page-link${aFlex}`}
          onClick={() => this.emitChange(page)}
          disabled={active}
        >
          {page}
        </button>
      </li>
    );
  }

  private generatePageButtons() {
    const buttons: any[] = [];
    const halfLimit = Math.floor((this.limit - 1) / 2);

    let startPage = Math.max(1, this.currentPage - halfLimit);
    let endPage = Math.min(this.totalPages, startPage + this.limit - 1);

    if (endPage - startPage < this.limit - 1) {
      startPage = Math.max(1, endPage - this.limit + 1);
    }

    if (startPage > 1) {
      buttons.push(this.renderPageButton(1));
      if (startPage > 2) buttons.push(this.renderEllipsis());
    }

    for (let i = startPage; i <= endPage; i++) {
      buttons.push(this.renderPageButton(i));
    }

    if (endPage < this.totalPages) {
      if (endPage < this.totalPages - 1) buttons.push(this.renderEllipsis());
      buttons.push(this.renderPageButton(this.totalPages));
    }

    return buttons;
  }

  render() {
    const sizeCls =
      this.size === 'sm' ? ' pagination-sm' : this.size === 'lg' ? ' pagination-lg' : '';
    const layoutCls =
      this.paginationLayout === 'center'
        ? ' justify-content-center flex-fill50'
        : this.paginationLayout === 'end'
        ? ' justify-content-end'
        : this.paginationLayout === 'fill'
        ? ' text-center'
        : '';
    const colorCls = this.paginationVariantColor ? ` ${this.paginationVariantColor}` : '';
    const plumageCls = this.plumage ? ' plumage' : '';

    const liFlex =
      this.paginationLayout === 'fill' ||
      this.paginationLayout === 'fill-left' ||
      this.paginationLayout === 'fill-right'
        ? ' flex-fill d-flex'
        : '';
    const aFlex =
      this.paginationLayout === 'fill' ||
      this.paginationLayout === 'fill-left' ||
      this.paginationLayout === 'fill-right'
        ? ' flex-grow-1'
        : '';

    return (
      <ul
        role="menubar"
        aria-disabled="false"
        aria-label="Pagination"
        class={`pagination b-pagination${sizeCls}${layoutCls}${colorCls}${plumageCls}`}
      >
        {!this.hideGotoEndButtons ? (
           <Fragment>
            <li
              role="presentation"
              aria-hidden={this.currentPage === 1 ? ('true' as any) : ('false' as any)}
              class={`page-item${this.currentPage === 1 ? ' disabled' : ''}${liFlex}`}
            >
              <button
                role="menuitem"
                type="button"
                tabIndex={this.currentPage === 1 ? -1 : 0}
                aria-label="Go to first page"
                aria-controls="pagination"
                class={`page-link${aFlex}`}
                onClick={this.firstPage}
                disabled={this.currentPage === 1}
              >
                {this.goToButtons === 'text' ? 'First' : '«'}
              </button>
            </li>
            <li
              role="presentation"
              aria-hidden={this.currentPage === 1 ? ('true' as any) : ('false' as any)}
              class={`page-item${this.currentPage === 1 ? ' disabled' : ''}${liFlex}`}
            >
              <button
                role="menuitem"
                type="button"
                tabIndex={this.currentPage === 1 ? -1 : 0}
                aria-label="Go to previous page"
                aria-controls="pagination"
                class={`page-link${aFlex}`}
                onClick={this.prevPage}
                disabled={this.currentPage === 1}
              >
                {this.goToButtons === 'text' ? 'Prev' : '‹'}
              </button>
            </li>
          </Fragment>
        ) : null}

        {this.generatePageButtons()}

        {!this.hideGotoEndButtons ? (
          <Fragment>
            <li
              role="presentation"
              aria-hidden={this.currentPage === this.totalPages ? ('true' as any) : ('false' as any)}
              class={`page-item${this.currentPage === this.totalPages ? ' disabled' : ''}${liFlex}`}
            >
              <button
                role="menuitem"
                type="button"
                tabIndex={this.currentPage === this.totalPages ? -1 : 0}
                aria-label="Go to next page"
                aria-controls="pagination"
                class={`page-link${aFlex}`}
                onClick={this.nextPage}
                disabled={this.currentPage === this.totalPages}
              >
                {this.goToButtons === 'text' ? 'Next' : '›'}
              </button>
            </li>
            <li
              role="presentation"
              aria-hidden={this.currentPage === this.totalPages ? ('true' as any) : ('false' as any)}
              class={`page-item${this.currentPage === this.totalPages ? ' disabled' : ''}${liFlex}`}
            >
              <button
                role="menuitem"
                type="button"
                tabIndex={this.currentPage === this.totalPages ? -1 : 0}
                aria-label="Go to last page"
                aria-controls="pagination"
                class={`page-link${aFlex}`}
                onClick={this.lastPage}
                disabled={this.currentPage === this.totalPages}
              >
                {this.goToButtons === 'text' ? 'Last' : '»'}
              </button>
            </li>
          </Fragment>
        ) : null}
      </ul>
    );
  }
}
