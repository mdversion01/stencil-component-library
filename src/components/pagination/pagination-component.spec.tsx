// src/components/pagination/pagination-component.spec.tsx
import { newSpecPage } from '@stencil/core/testing';
import { h } from '@stencil/core';
import { PaginationComponent } from './pagination-component';
import { StandardPagination } from './standard-pagination-component';
import { MinimizePagination } from './minimize-pagination-component';
import { ByPagePagination } from './by-page-pagination-component';

describe('pagination-component', () => {
  it('renders STANDARD by default', async () => {
    const page = await newSpecPage({
      components: [PaginationComponent, StandardPagination, MinimizePagination, ByPagePagination],
      template: () => <pagination-component currentPage={1} totalPages={10} paginationLayout="center" goToButtons="text" />,
    });
    expect(page.root).toMatchSnapshot();
    expect(page.root!.querySelector('standard-pagination-component')).not.toBeNull();
  });

  it('renders MINIMIZE when useMinimizePagination', async () => {
    const page = await newSpecPage({
      components: [PaginationComponent, StandardPagination, MinimizePagination, ByPagePagination],
      template: () => <pagination-component useMinimizePagination currentPage={2} totalPages={5} paginationLayout="end" size="sm" />,
    });
    expect(page.root).toMatchSnapshot();
    expect(page.root!.querySelector('minimize-pagination-component')).not.toBeNull();
  });

  it('renders BY-PAGE when useByPagePagination', async () => {
    const page = await newSpecPage({
      components: [PaginationComponent, StandardPagination, MinimizePagination, ByPagePagination],
      template: () => <pagination-component useByPagePagination currentPage={5} totalPages={42} paginationLayout="center" size="lg" plumage goToButtons="text" />,
    });
    expect(page.root).toMatchSnapshot();
    expect(page.root!.querySelector('by-page-pagination-component')).not.toBeNull();
  });

  it('size changer + display range snapshot', async () => {
    const page = await newSpecPage({
      components: [PaginationComponent, StandardPagination, MinimizePagination, ByPagePagination],
      template: () => (
        <pagination-component currentPage={1} totalRows={1234} showSizeChanger showDisplayRange pageSizeOptions={[10, 20, 50, 100, 'All']} paginationLayout="start" plumage />
      ),
    });
    expect(page.root).toMatchSnapshot();
    expect(page.root!.textContent).toMatch(/1-10 of 1234/);
  });

  it('emits page-changed when child emits change-page', async () => {
    const page = await newSpecPage({
      components: [PaginationComponent, StandardPagination, MinimizePagination, ByPagePagination],
      template: () => <pagination-component currentPage={1} totalPages={3} />,
    });
    const spy = jest.fn();
    page.root!.addEventListener('page-changed', (e: any) => spy(e.detail));

    const child = page.root!.querySelector('standard-pagination-component')!;
    child.dispatchEvent(new CustomEvent('change-page', { bubbles: true, detail: { page: 2 } }));
    await page.waitForChanges();

    expect(spy).toHaveBeenCalledWith({ page: 2, pageSize: 10 });
    expect(page.root!.getAttribute('current-page')).toBe('2');
  });

  it('warns when both variant flags set', async () => {
    const warn = jest.spyOn(console, 'warn').mockImplementation(() => {});
    const page = await newSpecPage({
      components: [PaginationComponent, StandardPagination, MinimizePagination, ByPagePagination],
      template: () => <pagination-component useMinimizePagination />, // start with one flag
    });
    // now toggle second flag to trigger @Watch
    page.root!.setAttribute('use-by-page-pagination', '');
    await page.waitForChanges();

    expect(warn).toHaveBeenCalledWith('[pagination] Both variants set; using minimize by precedence.');
    warn.mockRestore();
  });
});
