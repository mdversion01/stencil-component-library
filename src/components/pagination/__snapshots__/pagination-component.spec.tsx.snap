// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagination-component renders BY-PAGE when useByPagePagination 1`] = `
<pagination-component current-page="5" page-size="10" total-pages="42">
  <div class="d-flex pagination-layout plumage">
    <div class="center pagination-cell">
      <by-page-pagination-component limit="3" paginationvariantcolor="">
        <ul aria-disabled="false" aria-label="Pagination" class="b-pagination by-page justify-content-center pagination pagination-lg plumage" role="menubar">
          <li class="page-item" role="presentation">
            <button aria-controls="" aria-label="Go to first page" class="page-link" role="menuitem" tabindex="0" type="button">
              First
            </button>
          </li>
          <li class="page-item" role="presentation">
            <button aria-controls="" aria-label="Go to previous page" class="page-link" role="menuitem" tabindex="0" type="button">
              Prev
            </button>
          </li>
          <li aria-hidden="true" class="page-item" role="presentation">
            <div class="pages">
              Page
              <div aria-labelledby="pageNumberField" class="page-input-wrapper pl-input-container" role="presentation">
                <label class="sr-only" htmlfor="pageNumberField">
                  Page number
                </label>
                <input aria-label="Page number" aria-labelledby="paginationInputLabel paginationDescription" class="page-input pl-form-control pl-input-lg" id="paginationInput" inputmode="numeric" pattern="[0-9]*" type="text" value="5">
                <div class="b-underline" role="presentation">
                  <div aria-hidden="true" class="b-focus" role="presentation"></div>
                </div>
              </div>
              of 42
            </div>
          </li>
          <li class="page-item" role="presentation">
            <button aria-controls="" aria-label="Go to next page" class="page-link" role="menuitem" tabindex="0" type="button">
              Next
            </button>
          </li>
          <li class="page-item" role="presentation">
            <button aria-controls="" aria-label="Go to last page" class="page-link" role="menuitem" tabindex="0" type="button">
              Last
            </button>
          </li>
        </ul>
      </by-page-pagination-component>
    </div>
  </div>
</pagination-component>
`;

exports[`pagination-component renders MINIMIZE when useMinimizePagination 1`] = `
<pagination-component current-page="2" page-size="10" total-pages="5">
  <div class="d-flex pagination-layout">
    <div class="end pagination-cell">
      <minimize-pagination-component limit="3" paginationvariantcolor="">
        <ul aria-disabled="false" aria-label="Pagination" class="b-pagination justify-content-end pagination pagination-sm" role="menubar">
          <li class="page-item" role="presentation">
            <button aria-controls="" aria-label="Go to first page" class="page-link" role="menuitem" tabindex="0" type="button">
              «
            </button>
          </li>
          <li class="page-item" role="presentation">
            <button aria-controls="" aria-label="Go to previous page" class="page-link" role="menuitem" tabindex="0" type="button">
              ‹
            </button>
          </li>
          <li class="page-item" role="presentation">
            <button aria-controls="" aria-label="Go to next page" class="page-link" role="menuitem" tabindex="0" type="button">
              ›
            </button>
          </li>
          <li class="page-item" role="presentation">
            <button aria-controls="" aria-label="Go to last page" class="page-link" role="menuitem" tabindex="0" type="button">
              »
            </button>
          </li>
        </ul>
      </minimize-pagination-component>
    </div>
  </div>
</pagination-component>
`;

exports[`pagination-component renders STANDARD by default 1`] = `
<pagination-component current-page="1" page-size="10" total-pages="10">
  <div class="d-flex pagination-layout">
    <div class="center pagination-cell">
      <standard-pagination-component>
        <ul aria-disabled="false" aria-label="Pagination" class="b-pagination flex-fill50 justify-content-center pagination" role="menubar">
          <li aria-hidden="true" class="disabled page-item" role="presentation">
            <button aria-controls="pagination" aria-label="Go to first page" class="page-link" disabled="" role="menuitem" tabindex="-1" type="button">
              First
            </button>
          </li>
          <li aria-hidden="true" class="disabled page-item" role="presentation">
            <button aria-controls="pagination" aria-label="Go to previous page" class="page-link" disabled="" role="menuitem" tabindex="-1" type="button">
              Prev
            </button>
          </li>
          <li class="active page-item" role="presentation">
            <button aria-checked="true" aria-controls="pagination" aria-label="Go to page 1" class="page-link" disabled="" role="menuitemradio" tabindex="-1" type="button">
              1
            </button>
          </li>
          <li class="page-item" role="presentation">
            <button aria-checked="false" aria-controls="pagination" aria-label="Go to page 2" class="page-link" role="menuitemradio" tabindex="0" type="button">
              2
            </button>
          </li>
          <li class="page-item" role="presentation">
            <button aria-checked="false" aria-controls="pagination" aria-label="Go to page 3" class="page-link" role="menuitemradio" tabindex="0" type="button">
              3
            </button>
          </li>
          <li class="bv-d-xs-down-none disabled page-item" role="separator">
            <span class="page-link">
              ...
            </span>
          </li>
          <li class="page-item" role="presentation">
            <button aria-checked="false" aria-controls="pagination" aria-label="Go to page 10" class="page-link" role="menuitemradio" tabindex="0" type="button">
              10
            </button>
          </li>
          <li aria-hidden="false" class="page-item" role="presentation">
            <button aria-controls="pagination" aria-label="Go to next page" class="page-link" role="menuitem" tabindex="0" type="button">
              Next
            </button>
          </li>
          <li aria-hidden="false" class="page-item" role="presentation">
            <button aria-controls="pagination" aria-label="Go to last page" class="page-link" role="menuitem" tabindex="0" type="button">
              Last
            </button>
          </li>
        </ul>
      </standard-pagination-component>
    </div>
  </div>
</pagination-component>
`;

exports[`pagination-component size changer + display range snapshot 1`] = `
<pagination-component current-page="1" page-size="10" total-pages="1">
  <div class="pagination-split-layout plumage">
    <div class="pagination-cell start">
      <standard-pagination-component>
        <ul aria-disabled="false" aria-label="Pagination" class="b-pagination pagination plumage" role="menubar">
          <li aria-hidden="true" class="disabled page-item" role="presentation">
            <button aria-controls="pagination" aria-label="Go to first page" class="page-link" disabled="" role="menuitem" tabindex="-1" type="button">
              «
            </button>
          </li>
          <li aria-hidden="true" class="disabled page-item" role="presentation">
            <button aria-controls="pagination" aria-label="Go to previous page" class="page-link" disabled="" role="menuitem" tabindex="-1" type="button">
              ‹
            </button>
          </li>
          <li class="active page-item" role="presentation">
            <button aria-checked="true" aria-controls="pagination" aria-label="Go to page 1" class="page-link" disabled="" role="menuitemradio" tabindex="-1" type="button">
              1
            </button>
          </li>
          <li aria-hidden="true" class="disabled page-item" role="presentation">
            <button aria-controls="pagination" aria-label="Go to next page" class="page-link" disabled="" role="menuitem" tabindex="-1" type="button">
              ›
            </button>
          </li>
          <li aria-hidden="true" class="disabled page-item" role="presentation">
            <button aria-controls="pagination" aria-label="Go to last page" class="page-link" disabled="" role="menuitem" tabindex="-1" type="button">
              »
            </button>
          </li>
        </ul>
      </standard-pagination-component>
    </div>
    <div class="pagination-cell row-display">
      1-10 of 1234
    </div>
    <div class="end pagination-cell">
      <div class="size-changer">
        <label htmlfor="pageSize">
          Items per page:
        </label>
        <div aria-labelledby="selectField" class="pl-input-container" role="presentation">
          <select aria-invalid="false" aria-label="selectField" aria-labelledby="selectField" aria-multiselectable="false" class="form-control form-select" id="pageSize" role="listbox">
            <option selected="" value="10">
              10
            </option>
            <option value="20">
              20
            </option>
            <option value="50">
              50
            </option>
            <option value="100">
              100
            </option>
            <option value="All">
              All
            </option>
          </select>
          <div class="b-underline" role="presentation">
            <div aria-hidden="true" class="b-focus" role="presentation"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</pagination-component>
`;
